src/requests/utils.ts

```diff
@@ -0,0 +1,30 @@
+import * as querystring from 'querystring'
+
+export const getNormalizedEncodedURI = (uri: string) => {
+  const [pathname, search] = uri.split('?')
+  const escapedSearch = search ? querystring.escape(search) : ''
+
+  return encodeURI(escapedSearch ? `${pathname}?${escapedSearch}` : pathname)
+}
+
+const normalizeHeaderKey = (key: string) => key.toLowerCase().trim()
+
+export const getNormalizedHeaders = (
+  rawHeaders: Record<string, string | unknown>,
+  headersToNormalize: Array<string>
+) => {
+  //in case the list of the headers to normalize is not normalized itself
+  const normalizedHeadersToNormalize: string[] = headersToNormalize.map(normalizeHeaderKey)
+
+  const normalizedHeaders: Record<string, string> = {}
+
+  for (const [key, value] of Object.entries(rawHeaders)) {
+    const normalizedKey = normalizeHeaderKey(key)
+
+    if (normalizedHeadersToNormalize.includes(normalizedKey) && typeof value === 'string') {
+      normalizedHeaders[normalizedKey] = encodeURI(value.trim())
+    }
```

Comment:
If a header that should be signed is missing we should handle it somehow. We need to decide whether to throw or to just log it. 

Personally I feel like it should throw, and then an endpoint that uses this should return a bad request error

Reply:
I agree, but if we go with what @jelz suggested [here](https://github.com/contentful/node-apps-toolkit/pull/11#discussion_r512647521) and hence revert to initial implementation, that case wouldn't happen anymore (the list of headers is taken directly from the object you pass)